Entity refactor feature branch

Daiiii99 opened the pull request:
**This is a working-in-progress branch** opening a draft pr for visibility


Let's all use this branch as a feature branch and only merge it into main when it's working again.
To continue adding to this feature branch, do the following: 
1. checkout database-entity-refactor branch
2. create a new branch base on database-entity-refactor branch
3. after making changes, push the publish the new branch onto GitHub and use pull request to merge into database-entity-refactor branch
4. after everyone is done making their changes, we will merge database-entity-refactor into main
5. To avoid conflicts and headaches, DO NOT DIRECTLY PUSH INTO database-entity-refactor anymore, this applies to myself as well. If I want to make further adjustments, I will open a PR as well

Daiiii99 committed:
half done(replace class)

Daiiii99 committed:
Using classes

Daiiii99 committed:
Close function add

Daiiii99 committed:
Separate done

Daiiii99 committed:
To backfill changes from previous PRs (#40)

* 更新:
增加用户NickName登录系统
增加用户切换系统+姓名区别
增加cookie
增加一些特殊功能
优化界面

* Update:
1. Add comments to the index.jsp, coding with ease for other teammates
2. Improve the webpage, improve the user experience
   a.separate webpage with five domain
   b.alter the input box
   c.decorate the chatting box
3. Functionality
   a.Can login with the nickName
   b.Can change user with the nickName
4. Use the cookie in frontend. Refresh webpage will not override signin
5. Add test.html, for testing the UI webpage.

* Update:
1. Add comments to the index.jsp, coding with ease for other teammates
2. Improve the webpage, improve the user experience
   a.separate webpage with five domain
   b.alter the input box
   c.decorate the chatting box
3. Functionality
   a.Can login with the nickName
   b.Can change user with the nickName
4. Use the cookie in frontend. Refresh webpage will not override signin
5. Add test.html, for testing the UI webpage.

* 1. Package the function. Make code more readable.
2. Add extra comments and alerts
3. Add emoji tab, user can tap emoji in the input box
4. Switch the message order, the latest message will now appear at the top
5. Adding the icomoon font size to the webpage, better design

* Add web art design manuscripts at images file.

* 1. Design the project
2. Add the Vue

* 1. Add Vue

* 1. Use Vue to realize the post history display
2. Use Vue to realize the displaying new post after clicking the sending message
3. Delete most part of the jQuery code
4. Add more comments, arrange all the click event at one place, making code more readable
5. Add servlet05, servlet06, one for receive the new message, one for response the new message from the frontend.

* 1. Arrange the code, make more readable
2. Debugging, user's own posts will have the delete button

* Adding more functions

* Add all controller, presenter and gateway for our project.

* Add all controller, presenter and gateway for our project.

* Add two presenters, modified user register gateway and use case.

* Revert "Revert "added pop up windows and changed 'like' related functions""

This reverts commit 51f47598fb5233ca760d590832471070def5c1a0.

* Adding more functions

* 1. Add user head_pic system, right now generated randomly.
2. Add the CSS file, make the code more readable

* 1. Using Vue to realize login and logout

* 1. Adding liked method
2. Adding header function for frontend

* changed (#28)

* 1. Cleaning

* 1. Adding reply to frontend
2. Able to send reply

* Wrote or updated the controller, gateway, presenter and use case for the following functionalities:
1) like post
2) delete post
3) login
4) post post
5) register
6) send reply

* Improve the code for the frontend:
1. Add three classes (objects) for fontend: post class, user class, post list class
2. Add two tests for post class and user class
3. Increase the security level, (a)user is not able to change the user recorded, unless change the cookie, but requires a proper password when reloading.
 (b)User is not able to change the id of any post
 (c) User is not able to change the structure of any class
4. Add icon for the web apps

* made some small change

* Add comments to the use cases, presenters, controllers and gateways.

* 1. Check the code again
2. Add some comments
3. Complete the login, reply message function
4. Enhance the coding readability

* 1.Prevent user alter the cookie

* Fix the bugs associated with reply function.
1. User can see their sending reply immediately
2. Click the reply box will trigger the comment event
3. Add the max-height for post content.
4. Add word-break for reply content
5. The background changes when user open the specific reply

* 1.Fix the reply function bugs

* Add comments to the use cases, presenters, controllers and gateways.

* Add comments to the use cases, presenters, controllers and gateways.

* Add comments to the use cases, presenters, controllers and gateways.

* Add comments to the use cases, presenters, controllers and gateways.

* Added a read me.

Co-authored-by: mugengliu <71624274+mugengliu@users.noreply.github.com>
Co-authored-by: MikeJiahaoXu <mikeke.xu@mail.utoronto.ca>
Co-authored-by: ruoxinhe <ruoxin.he@mail.utoronto.ca>
Co-authored-by: MikeJiahaoXu <113255498+MikeJiahaoXu@users.noreply.github.com>

mugengliu commented:
Does it work right now?

MikeJiahaoXu reviewed (changes requested):
1, make sure all names in entity and database are using the correct java naming convention.
2, you can delete the methods that you think will not be used in entities.

Daiiii99 committed:
Name changed and deletion done

Daiiii99 committed:
Refactor after forced push (#45)

* 更新:
增加用户NickName登录系统
增加用户切换系统+姓名区别
增加cookie
增加一些特殊功能
优化界面

* Update:
1. Add comments to the index.jsp, coding with ease for other teammates
2. Improve the webpage, improve the user experience
   a.separate webpage with five domain
   b.alter the input box
   c.decorate the chatting box
3. Functionality
   a.Can login with the nickName
   b.Can change user with the nickName
4. Use the cookie in frontend. Refresh webpage will not override signin
5. Add test.html, for testing the UI webpage.

* Update:
1. Add comments to the index.jsp, coding with ease for other teammates
2. Improve the webpage, improve the user experience
   a.separate webpage with five domain
   b.alter the input box
   c.decorate the chatting box
3. Functionality
   a.Can login with the nickName
   b.Can change user with the nickName
4. Use the cookie in frontend. Refresh webpage will not override signin
5. Add test.html, for testing the UI webpage.

* 1. Package the function. Make code more readable.
2. Add extra comments and alerts
3. Add emoji tab, user can tap emoji in the input box
4. Switch the message order, the latest message will now appear at the top
5. Adding the icomoon font size to the webpage, better design

* Add web art design manuscripts at images file.

* 1. Design the project
2. Add the Vue

* 1. Add Vue

* 1. Use Vue to realize the post history display
2. Use Vue to realize the displaying new post after clicking the sending message
3. Delete most part of the jQuery code
4. Add more comments, arrange all the click event at one place, making code more readable
5. Add servlet05, servlet06, one for receive the new message, one for response the new message from the frontend.

* 1. Arrange the code, make more readable
2. Debugging, user's own posts will have the delete button

* Adding more functions

* Add all controller, presenter and gateway for our project.

* Add all controller, presenter and gateway for our project.

* Add two presenters, modified user register gateway and use case.

* Revert "Revert "added pop up windows and changed 'like' related functions""

This reverts commit 51f47598fb5233ca760d590832471070def5c1a0.

* Adding more functions

* 1. Add user head_pic system, right now generated randomly.
2. Add the CSS file, make the code more readable

* 1. Using Vue to realize login and logout

* 1. Adding liked method
2. Adding header function for frontend

* changed (#28)

* 1. Cleaning

* 1. Adding reply to frontend
2. Able to send reply

* Wrote or updated the controller, gateway, presenter and use case for the following functionalities:
1) like post
2) delete post
3) login
4) post post
5) register
6) send reply

* Improve the code for the frontend:
1. Add three classes (objects) for fontend: post class, user class, post list class
2. Add two tests for post class and user class
3. Increase the security level, (a)user is not able to change the user recorded, unless change the cookie, but requires a proper password when reloading.
 (b)User is not able to change the id of any post
 (c) User is not able to change the structure of any class
4. Add icon for the web apps

* made some small change

* Add comments to the use cases, presenters, controllers and gateways.

* 1. Check the code again
2. Add some comments
3. Complete the login, reply message function
4. Enhance the coding readability

* 1.Prevent user alter the cookie

* Fix the bugs associated with reply function.
1. User can see their sending reply immediately
2. Click the reply box will trigger the comment event
3. Add the max-height for post content.
4. Add word-break for reply content
5. The background changes when user open the specific reply

* 1.Fix the reply function bugs

* Add comments to the use cases, presenters, controllers and gateways.

* Add comments to the use cases, presenters, controllers and gateways.

* Add comments to the use cases, presenters, controllers and gateways.

* Add comments to the use cases, presenters, controllers and gateways.

* Added a read me.

Co-authored-by: mugengliu <71624274+mugengliu@users.noreply.github.com>
Co-authored-by: MikeJiahaoXu <mikeke.xu@mail.utoronto.ca>
Co-authored-by: ruoxinhe <ruoxin.he@mail.utoronto.ca>
Co-authored-by: MikeJiahaoXu <113255498+MikeJiahaoXu@users.noreply.github.com>

MikeJiahaoXu committed:
1, Improved java naming conventions (#47)

2, Made change to the static methods, so they are now non-static.
3, Implemented all tests for use cases.
4, Combined the overly used methods for checking username and password to get rid of the duplicate codes.
5, Used the newest version of database methods.

Daiiii99 committed:
Add dependency

Daiiii99 committed:
Servlet refactor uses entity classes

Daiiii99 committed:
Bug fix

Daiiii99 committed:
fix typo

Daiiii99 committed:
frontend refactor

Daiiii99 committed:
Merge branch 'main' into database-entity-refactor
